app/connectors/GetBusinessDetailsConnector.scala

```diff
@@ -30,13 +30,24 @@ class GetBusinessDetailsConnector @Inject()(val http: HttpClient,
                                             val appConfig: MicroserviceAppConfig
                                            )(implicit ec: ExecutionContext) extends RawResponseReads {
 
-  def getBusinessDetailsUrl(nino: String): String = s"${appConfig.ifUrl}/registration/business-details/nino/$nino"
+  def getBusinessDetailsUrl(nino: String): String = {
+    s"${appConfig.ifUrl}/registration/business-details/nino/$nino"
+  }
+
+  def getIncomeSourceDetailsUrl(mtdRef: String): String = {
+    val idtype = "mtdId"
+    s"${appConfig.ifUrl}/registration/business-details/$idtype/$mtdRef"
+  }
 
   def headers: Seq[(String, String)] = appConfig.getIFHeaders("1171")
 
-  def getBusinessDetails(nino: String)(implicit headerCarrier: HeaderCarrier): Future[IncomeSourceDetailsResponseModel] = {
+  def getBusinessDetails(ninoOrMtdRef: String, accessType: BusinessDetailsAccessType)
+                        (implicit headerCarrier: HeaderCarrier): Future[IncomeSourceDetailsResponseModel] = {
 
-    val url = getBusinessDetailsUrl(nino)
+    val url = accessType match {
```

Comment:
lines 47-50 is a good candidate to move these into separate method instead of having: getBusinessDetailsUrl + getIncomeSourceDetailsUrl